[Security Solution][Detection Engine] ML Cypress Improvements

rylnd opened the pull request:
## Summary

This PR contains a collection of fixes, mainly cherry-picked from #181926. Since some of these tests are failing sporadically on main, I'm opening this "test fix" PR for expedience while #181926 is still in review.


### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
    * [Cypress * 100](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6385)
    * [Serverless Cypress * 100](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6386)

rylnd committed:
Address flakiness in ML cypress test

This commit contains a few improvements, cherry-picked from #181926.
Full context/details can be found there, but I'll try to summarize
below:

* ML form helpers receive normal rule parameters (instead of humanized
  job names)
* Better starting state: deletes rules and stops ML jobs before test
  suite
* Relaxed assertions: removes assertions that fail if there's more than
  one test in the environment (this is somewhat redundant with the
  previous improvement, but no harm in having both).

rylnd commented:
/ci

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#6383](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6383)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/configs/ess.config.ts: 79/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#6384](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6384)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/configs/serverless.config.ts: 72/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

rylnd commented:
/ci

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#6381](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6381)
[✅] Security Solution Detection Engine - Cypress: 100/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#6382](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6382)
[✅] [Serverless] Security Solution Detection Engine - Cypress: 100/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#6385](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6385)
[✅] Security Solution Detection Engine - Cypress: 100/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#6386](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6386)
[✅] [Serverless] Security Solution Detection Engine - Cypress: 100/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F186831%2Fhead)

dhurley14 commented on x-pack/test/security_solution_cypress/cypress/support/machine_learning.ts:
```diff
@@ -0,0 +1,92 @@
+ * @param moduleName the name of the ML module to set up
+ * @returns the response from the setup module request
+ */
+export const executeSetupModuleRequest = ({ moduleName }: { moduleName: string }) =>
```
not seeing this used in this PR? You mentioned you cherry picked some things so maybe this is just unused but figured I'd bring this up.

dhurley14 commented on x-pack/test/security_solution_cypress/cypress/support/machine_learning.ts:
```diff
@@ -0,0 +1,92 @@
+ * @param jobIds the job IDs for which to force start datafeeds
+ * @returns the response from the force start datafeeds request
+ */
+export const forceStartDatafeeds = ({ jobIds }: { jobIds: string[] }) =>
```
FYI: Not seeing this used either.

dhurley14 commented on x-pack/test/security_solution_cypress/cypress/support/machine_learning.ts:
```diff
@@ -0,0 +1,92 @@
+ * @param jobIds the job IDs for which to stop datafeeds
+ * @returns the response from the stop datafeeds request
+ */
+export const stopDatafeeds = ({ jobIds }: { jobIds: string[] }) =>
```
FYI not seeing this used either

dhurley14 reviewed (commented):
There are three functions added here that aren't used in the changes. Not sure if we plan on using these in the future but maybe we should remove them from this PR until they are utilized.

rylnd committed:
Remove unused helper methods

These were cherry-picked from the ML suppression branch, where they're
used more fully. For now, we only use forceStopAndCloseJob to get a
clean test state.

rylnd commented:
@dhurley14 I removed those unused helpers from this PR 👍

dhurley14 reviewed (approved):
👍

rylnd committed:
Merge branch 'main' into ml_cypress_improvements
